fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Lao.java

```diff
@@ -79,6 +79,83 @@ public Lao(String name, PublicKey organizer, long creation) {
+    if (lao.modificationId != null) {
+      this.modificationId = new MessageID(lao.modificationId);
+    }
+
+    this.witnesses = lao.witnesses.stream().map(PublicKey::new).collect(Collectors.toSet());
+    this.witnessMessages =
+        lao.witnessMessages.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    entry -> new MessageID(entry.getKey()),
+                    entry -> new WitnessMessage(entry.getValue())));
+    this.pendingUpdates =
+        lao.pendingUpdates.stream().map(PendingUpdate::new).collect(Collectors.toSet());
+    this.rollCalls =
+        lao.rollCalls.entrySet().stream()
+            .collect(Collectors.toMap(Map.Entry::getKey, entry -> new RollCall(entry.getValue())));
+    this.elections =
+        lao.elections.entrySet().stream()
+            .collect(Collectors.toMap(Map.Entry::getKey, entry -> new Election(entry.getValue())));
+    this.allChirps =
+        lao.allChirps.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    entry -> new MessageID(entry.getKey()), entry -> new Chirp(entry.getValue())));
+    this.chirpsByUser =
+        lao.chirpsByUser.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    entry -> new PublicKey(entry.getKey()),
+                    entry ->
+                        entry.getValue().stream()
+                            .map(MessageID::new)
+                            .collect(Collectors.toList())));
+    this.messageIdToElectInstance =
+        lao.messageIdToElectInstance.entrySet().stream()
+            .collect(Collectors.toMap(entry -> new MessageID(entry.getKey()), Map.Entry::getValue));
+    this.keyToNode =
+        lao.keyToNode.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    entry -> new PublicKey(entry.getKey()),
+                    entry -> new ConsensusNode(entry.getValue())));
+    this.pubKeyByHash =
+        lao.pubKeyByHash.entrySet().stream()
+            .collect(Collectors.toMap(Map.Entry::getKey, entry -> new PublicKey(entry.getValue())));
+    this.transactionHistoryByUser =
+        lao.transactionHistoryByUser.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    entry -> new PublicKey(entry.getKey()),
+                    entry ->
+                        entry.getValue().stream()
+                            .map(TransactionObject::new)
+                            .collect(Collectors.toList())));
+    this.transactionByUser =
+        lao.transactionByUser.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    entry -> new PublicKey(entry.getKey()),
+                    entry ->
+                        entry.getValue().stream()
+                            .map(TransactionObject::new)
+                            .collect(Collectors.toList())));
```

Comment:
this code and similar ones elsewhere are kind of scary / ugly. I believe I understand the general idea, but is there no cleaner / more readable / more succinct way of doing this ? 🤔

Reply:
Unfortunately there is no deep copier of `Map` in java (from my search on S.O). 
I thought about making a custom function for that but the problem is that the `Object` class does not have a copy constructor in the interface. So I imagined adding an abstract type that does but it seemed a bit overkill for a few copy constructors.
Furthermore, we plan on breaking down the Lao object later so it should be less appalling 😄 

Finally the stream syntax might look a bit scary / ugly but the alternative is extremly verbose so I settled on that, Alternative without stream for line 148 to 156 looks like this just in case

````
```
    HashMap<PublicKey, List<TransactionObject>> copy = new HashMap<>();
    for (Map.Entry<PublicKey, List<TransactionObject>> entry : lao.transactionByUser.entrySet()){
      List<TransactionObject> listCopy = new ArrayList<>();
      for (TransactionObject transactionObject : entry.getValue()){
        listCopy.add(new TransactionObject(transactionObject));
      }
      copy.put(new PublicKey(entry.getKey()), listCopy);
    }
    this.transactionByUser = copy;
```
````
It's the same amount of lines but it is much more packed.

Reply:
This was how I saw things when implementing it, if someone has a better idea I'll gladly take it 😄

Reply:
Okay, I understand better.
I understand Lao being too big, and this makes it pretty obvious.
But if I recall correctly, the election object is almost as ugly.

I'd also support a `Copiable` interface, as it seems a bit weird for some classes constructor to have to deal with how different objects need to be copied.

Reply:
Okay, I understand better.
I understand Lao being too big, and this makes it pretty obvious.
But if I recall correctly, the election object is almost as ugly.

I'd also support a `Copiable` interface, as it seems a bit weird for some classes constructor to have to deal with how different objects need to be copied.